tensorflow/compiler/mlir/xla/transforms/xla_legalize_to_linalg.cc

```diff
@@ -192,6 +192,110 @@ class ScalarPointwiseToStandardConverter : public OpConversionPattern<LhloOp> {
   }
 };
 
+//===----------------------------------------------------------------------===//
+// xla_lhlo.convolution conversion pattern.
+//===----------------------------------------------------------------------===//
+
+/// Converts xla_lhlo.convolution operation to a linalg.conv op.
+struct ConvToLinalgConverter : public OpConversionPattern<xla_lhlo::ConvOp> {
+ public:
+  using OpConversionPattern<xla_lhlo::ConvOp>::OpConversionPattern;
+
+  //  This code has been adapted from IREE's
+  //  (https://github.com/google/iree/) xla_hlo -> linalg conversion.
+  LogicalResult matchAndRewrite(
+      xla_lhlo::ConvOp op, ArrayRef<Value> args,
+      ConversionPatternRewriter& rewriter) const final {
+    // If there is explicit ordering info for the dimensions, propagate that
+    // through.
```

Comment:
The comment seems to indicate that this if block would "propagate" something, isn't it just checking some validity criteria?

Reply:
May be the phrasing was misleading. It's the attributes that are being propagated from xla_lhlo to linalg. The conversion still succeeds in the absence of that information.

Reply:
Rephrased this a bit.